server/src/main/java/org/apache/iotdb/db/mpp/operator/meta/MetaScanOperator.java

```diff
@@ -0,0 +1,120 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.operator.meta;
+
+import org.apache.iotdb.commons.consensus.ConsensusGroupId;
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.metadata.schemaregion.SchemaEngine;
+import org.apache.iotdb.db.metadata.schemaregion.SchemaRegion;
+import org.apache.iotdb.db.mpp.operator.Operator;
+import org.apache.iotdb.db.mpp.operator.OperatorContext;
+import org.apache.iotdb.tsfile.read.common.block.TsBlock;
+
+import java.io.IOException;
+import java.util.List;
+
+public class MetaScanOperator implements Operator {
+
+  protected OperatorContext operatorContext;
+  protected TsBlock tsBlock;
+  private boolean hasCachedTsBlock;
+
+  protected SchemaRegion schemaRegion;
+  protected int limit;
+  protected int offset;
+  protected PartialPath partialPath;
+  protected boolean isPrefixPath;
+  protected List<String> columns;
+
+  protected MetaScanOperator(
+      OperatorContext operatorContext,
+      ConsensusGroupId schemaRegionId,
+      int limit,
+      int offset,
+      PartialPath partialPath,
+      boolean isPrefixPath,
+      List<String> columns) {
+    this.operatorContext = operatorContext;
+    this.schemaRegion = SchemaEngine.getInstance().getSchemaRegion(schemaRegionId);
+    this.limit = limit;
+    this.offset = offset;
+    this.partialPath = partialPath;
+    this.isPrefixPath = isPrefixPath;
+    this.columns = columns;
+  }
+
+  protected TsBlock createTsBlock() throws MetadataException {
+    return null;
+  }
+
+  public PartialPath getPartialPath() {
+    return partialPath;
+  }
+
+  public SchemaRegion getSchemaRegion() {
+    return schemaRegion;
+  }
+
+  public int getLimit() {
+    return limit;
+  }
+
+  public int getOffset() {
+    return offset;
+  }
+
+  public void setLimit(int limit) {
+    this.limit = limit;
+  }
+
+  public void setOffset(int offset) {
+    this.offset = offset;
+  }
+
+  public boolean isPrefixPath() {
+    return isPrefixPath;
+  }
+
+  @Override
+  public OperatorContext getOperatorContext() {
+    return operatorContext;
+  }
+
+  @Override
+  public TsBlock next() throws IOException {
+    hasCachedTsBlock = false;
+    return tsBlock;
+  }
+
+  @Override
+  public boolean hasNext() throws IOException {
+    try {
+      if (tsBlock == null) {
+        tsBlock = createTsBlock();
+        if (tsBlock.getPositionCount() > 0) {
+          hasCachedTsBlock = true;
+        }
+      }
+      return hasCachedTsBlock;
+    } catch (MetadataException e) {
+      throw new IOException(e);
```

Comment:
`MetadataException` is not an `IOException`, just wrap it with `IOException` is not very good. You can throw a `RuntimeException` if you don't want to catch this `MetadataException` outside the Operator and do something special to handle this kind of Execption.

Reply:
And actually, we should not let `next()` and `hasNext()` method in `Operator` throw `IOException`, I will fix that in another pr.

Reply:
fixed